api/src/main/scala/cryptonite/ProductsService.scala

```diff
@@ -26,53 +26,43 @@ class ProductsService(gateways: List[Gateway], portfolioService: PortfolioServic
       booksByGateway <- filteredGateways.traverse(g => EitherT(getBooks(g)))
     } yield {
       val books = booksByGateway.flatten
-      val filteredBooks = books.filter(b => acceptBase(b,base)).filter(b => acceptQuote(b,quote))
+      val filteredBooks = books
+        .filter(b => acceptBase(b,base))
+        .filter(b => acceptQuote(b,quote))
       val sortedBooks = sortBooks(filteredBooks, sortBy)
       order(sortedBooks, ascending)
     }).value
 
   }
 
-  def getBooks(gateway: Gateway): Future[Either[ApiError, List[Book]]] = {
+  def getBooks(gateway: Gateway): Future[Either[ApiError, List[Book]]] =
     (for {
       tickers <- EitherT(gateway.read())
       amounts <- tickers.traverse(t => EitherT(portfolioService.get(t.product.base.index.code)))
     } yield {
-      val books = tickers.zip(amounts).map { case (t, a) => createBook(gateway.exchange(), t, a) }
-      books
+      tickers.zip(amounts).map { case (t, a) => createBook(gateway.exchange(), t, a) }
     }).value
-  }
 
-  def acceptBase(book: Book, base: Option[Currency]): Boolean = {
-    base match {
-      case Some(currency) => if (book.product.base == currency) true else false
-      case None => true
-    }
-  }
+  def acceptBase(book: Book, base: Option[Currency]): Boolean =
+    base.forall{ b => if (book.product.base == b) true else false }
```

Comment:
Prova con `base.forall(_ == book.product.base)`

Reply:
In scala `if` è un'espressione, quindi:
`if (condition) true else false` è equivalente a scrivere semplicemente `condition`.